infra/app/service/notifications.tf

```diff
@@ -0,0 +1,46 @@
+locals {
+  # If this is a temporary environment, re-use an existing email identity. Otherwise, create a new one.
+  domain_identity_arn = module.app_config.enable_notifications ? (
+    !local.is_temporary ?
+    module.notifications_email_domain[0].domain_identity_arn :
+    module.existing_notifications_email_domain[0].domain_identity_arn
+  ) : null
+  notifications_environment_variables = module.app_config.enable_notifications ? {
+    AWS_PINPOINT_APP_ID       = module.notifications[0].app_id,
+    AWS_PINPOINT_SENDER_EMAIL = local.notifications_config.sender_email
+  } : {}
+  notifications_app_name = "${local.prefix}${local.notifications_config.name}"
+}
+
+# If the app has `enable_notifications` set to true AND this is not a temporary
+# environment, then create a email notification identity.
+module "notifications_email_domain" {
+  count  = module.app_config.enable_notifications && !local.is_temporary ? 1 : 0
+  source = "../../modules/notifications-email-domain/resources"
+
+  domain_name         = local.service_config.domain_name
+  sender_email        = local.notifications_config.sender_email
+  sender_display_name = local.notifications_config.sender_display_name
```

Comment:
I don't think these variables are needed

Reply:
Wow... these module signatures are looking clean as heck

Reply:
:) haha that's basically the thing I care about the most when it comes to the platform design. It's easier to add things to interfaces than change or remove them so I like to keep them as minimal as possible.